server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/UserRoutes.java

```diff
@@ -164,6 +213,21 @@ public void defineAllowedFromHeaders() {
             jsonTransformer);
     }
 
+    private List<UserResponse> getUsers(Request request, Response response) throws UsersRepositoryException {
+        List<UserCondition> conditions = mapCondition.entrySet()
+            .stream()
+            .filter(entry -> request.queryParams().contains(entry.getKey()))
+            .map(Map.Entry::getValue)
+            .toList();
```

Comment:
What happens if I supply an unknown parameter?

Reply:
Maybe we can ignore case?

Reply:
It would be more idiomatic to reduce the conditions into one:

```
    interface UserCondition {
        static UserCondition ALL = any -> true;
        
        boolean satisfyCondition(UserResponse user);

        default UserCondition and(UserCondition other) {
            return user -> satisfyCondition(user) && other.satisfyCondition(user);
        }
    }
```

so that we can `.reduce(UserCOndition.ALL, UserCondition::and)`